src/icalendar/cal/event.py

```diff
@@ -386,6 +393,106 @@ def end(self, end: date | datetime | None):
         """Set the end."""
         self.DTEND = end
 
+    def set_duration(
```

Comment:
Hi, I wonder why this turns up in this PR. I think that was part of another PR?
If working on the others would get this into main, then I will look at those first.